viamonvif/service.go

```diff
@@ -287,6 +319,58 @@ func downloadPreviewImage(ctx context.Context, logger logging.Logger, snapshotUR
 	return dataURL, nil
 }
 
+// fetchImageFromRTSPURL fetches the image from the rtsp URL and returns it as a data URL.
+func fetchImageFromRTSPURL(ctx context.Context, logger logging.Logger, rtspURL string) (string, error) {
+	// Wrap viamrtsp.Config in a resource.Config
+	rtspConfig := viamrtsp.Config{
+		Address: rtspURL,
+	}
+	resourceConfig := resource.Config{
+		Name:                "tmp-camera",
+		API:                 camera.API,
+		Model:               viamrtsp.ModelAgnostic,
+		ConvertedAttributes: &rtspConfig,
+	}
+
+	camera, err := viamrtsp.NewRTSPCamera(ctx, nil, resourceConfig, logger)
+	if err != nil {
+		return "", fmt.Errorf("failed to create RTSP camera: %w", err)
+	}
+	defer func() {
+		if closeErr := camera.Close(ctx); closeErr != nil {
+			logger.Warnf("failed to close camera: %v", closeErr)
+		}
+	}()
+
+	retryInterval := rtspImageInterval
+	timeout := rtspPollTimeout
+	ticker := time.NewTicker(retryInterval)
+	defer ticker.Stop()
+	timeoutChan := time.After(timeout)
+	var imageErr error
+	if err := ctx.Err(); err != nil {
+		return "", fmt.Errorf("context already canceled before fetching image: %w", err)
```

Comment:
This check is not necessary. Just add the check (with an appropriate error) inside the loop before the select.

Reply:
Ok, yes removed extra check. I think the ` <-ctx.Done()` case should handle things.